pageserver/src/tenant/timeline.rs

```diff
@@ -3270,19 +2934,32 @@ impl Timeline {
         //    size length. Compaction will likely create the same set of n files afterwards.
         //
         // This failpoint is a superset of both of the cases.
-        fail_point!("compact-level0-phase1-return-same", |_| {
-            println!("compact-level0-phase1-return-same"); // so that we can check if we hit the failpoint
-            Ok(CompactLevel0Phase1Result {
-                new_layers: level0_deltas
-                    .iter()
-                    .map(|x| x.clone().downcast_delta_layer().unwrap())
-                    .collect(),
-                deltas_to_compact: level0_deltas
-                    .iter()
-                    .map(|x| x.layer_desc().clone().into())
-                    .collect(),
-            })
-        });
+        if cfg!(feature = "testing") {
+            // FIXME: utils does not depend on `fail` so there's no non-macro answer to this
```

Comment:
I could now move this "is_failpoint_enabled" to `failpoint_support`. We have some other nagging stuff as well.

Reply:
Well, or not.